tests/test_executable.py

```diff
@@ -238,3 +247,118 @@ def test_load_config_with_state_arg(self):
             cfg['patchworks']
         )
         self.assertEqual('some_value', cfg['some_other_state'])
+
+    def test_save_state(self):
+        """Ensure save_state works."""
+        def merge_two_dicts(dict1, dict2):
+            """Given two dicts, merge them into json_data new dict as
+            json_data shallow copy."""
+            result = dict1.copy()
+            result.update(dict2)
+            return result
+
+        # cfg has to have 'state', otherwise it returns None
+        self.assertIsNone(executable.save_state({}, {}))
+
+        config_file = []
+        args = ['--rc', '/tmp/testing.ini', '--workdir', '/tmp/workdir',
+                '--state', 'run', '--runner', 'myrunner', '[\'value\']']
```

Comment:
Runer still takes a json, so instead of `'[\'value\']'`, there should be something like `'{"key": "value"}'`so the test is more accurate.